### PR TITLE
Added drag and drop support for shader include files in shader editor

### LINKED ISSUE #74392: Drag-dropping an .gdshaderinc into the shader editor does not open the shader
### Godot version

4.0

### System information

Fedora Linux 37, Nvidia 4090, Vulkan 1.3, nv 525.60.11

### Issue description

Dropping a .gdshaderinc file from the FileSystem tab into the Shader Editor tab does not open the file like is done when a .gdshader is dropped

### Steps to reproduce

Create a .gdshaderinc file and drop it into the Shader Editor

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by fahadshihab**
This is in reference to issue #74392. 

_Production edit: Fixes https://github.com/godotengine/godot/issues/74392_

### CODE REVIEW

**Review by clayjohn — APPROVED** [review]
Looks good! Tested locally and it works

File: editor/plugins/shader_editor_plugin.cpp [hunk 1]
```diff
@@ -405,11 +411,13 @@ void ShaderEditorPlugin::drop_data_fw(const Point2 &p_point, const Variant &p_da
...
 			String file = files[i];
-			if (!ResourceLoader::exists(file, "Shader")) {
-				continue;
+			Ref<Resource> res;
+			if (ResourceLoader::exists(file, "Shader")) {
+				res = ResourceLoader::load(file);
+			}
+			if (ResourceLoader::exists(file, "ShaderInclude")) {
+				res = ResourceLoader::load(file);
 			}
```
Comment (YuriSizov) [inline editor/plugins/shader_editor_plugin.cpp:420]:
```suggestion
			if (ResourceLoader::exists(file, "Shader") || ResourceLoader::exists(file, "ShaderInclude")) {
				res = ResourceLoader::load(file);
			}
```

Comment (YuriSizov) [reply editor/plugins/shader_editor_plugin.cpp:420]: When you apply the suggestion, make sure to squash the commits. We prefer all PRs to contain only a single commit in them. You can keep the current commit message as the final message.

File: editor/plugins/shader_editor_plugin.cpp [hunk 2]
```diff
@@ -405,13 +411,11 @@ void ShaderEditorPlugin::drop_data_fw(const Point2 &p_point, const Variant &p_da
...
-			}
-
-			Ref<Resource> res = ResourceLoader::load(file);
-			if (res.is_valid()) {
-				edit(res.ptr());
+			if (ResourceLoader::exists(file, "Shader") || ResourceLoader::exists(file, "ShaderInclude")) {
+				Ref<Resource> res = ResourceLoader::load(file);
+				if (res.is_valid()) {
+					edit(res.ptr());
+				}
```
Comment (YuriSizov) [inline editor/plugins/shader_editor_plugin.cpp:418]: There is no need to nest everything like that, this was fine before. Generally prefer to keep nesting to the minimum, it makes code harder to read.

**Comment by YuriSizov** [comment]
Thanks, and congrats on your first merged pull request!

**Comment by fahadshihab** [comment]
Thanks! @YuriSizov and @clayjohn

**Pull request merged**